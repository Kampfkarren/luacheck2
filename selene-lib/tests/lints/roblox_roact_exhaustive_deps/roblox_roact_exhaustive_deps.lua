local React

local notreactive1 = {}

local function Component1()
    local reactive1 = {}
    local reactive2 = {}

    React.useEffect(function()
        local allowed = notreactive1
        local notallowed = reactive1
        print(reactive2)
    end, {})
end

local function Component2()
    local reactive1 = {}

    React.useEffect(function()
        allowed(reactive1, `{notreactive1}`)
    end, { reactive1 })
end

local function Component3()
    local reactive1 = {}
    local reactive2 = {}

    React.useEffect(function()
        notreactive1(reactive1(reactive2()))
    end, depArray(reactive1))
end

local function Component4()
    local _, setState = React.useState()

    React.useEffect(function()
        setState()
    end, {})
end

local function Component5()
    local reactive1 = {}

    React.useEffect(function()
        local notreactive = function()
            if true then
                local notallowed = reactive1
            end
        end
        notreactive()
    end, {})
end

local function MakeComponent()
    local notreactive = {}

    local function Component()
        local reactive1 = {}

        React.useEffect(function()
            local allowed = notreactive
            local allowed = reactive1
        end, { reactive1 })
    end
end
