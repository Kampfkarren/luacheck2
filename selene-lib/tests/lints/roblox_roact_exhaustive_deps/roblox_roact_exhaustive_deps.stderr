error[roblox_roact_exhaustive_deps]: React hook useEffect has missing dependencies: 'reactive1' and 'reactive2'. Either include them or remove the dependency array.
   ┌─ roblox_roact_exhaustive_deps.lua:13:10
   │
13 │     end, {})
   │          ^^

error[roblox_roact_exhaustive_deps]: React hook useEffect has a missing dependency: 'reactive2'. Either include it or remove the dependency array.
   ┌─ roblox_roact_exhaustive_deps.lua:30:10
   │
30 │     end, depArray(reactive1))
   │          ^^^^^^^^^^^^^^^^^^^

error[roblox_roact_exhaustive_deps]: React hook useEffect has a missing dependency: 'reactive1'. Either include it or remove the dependency array.
   ┌─ roblox_roact_exhaustive_deps.lua:51:10
   │
51 │     end, {})
   │          ^^

